client/app/bundles/course/assessment/question/programming/actions/onlineEditorActionCreators.jsx

```diff
@@ -33,25 +33,81 @@ export function deletePythonTestCase(testType, index) {
   };
 }
 
-export function updateNewDataFile(filename, index) {
+export function updatePythonNewDataFile(filename, index) {
   return {
     type: actionTypes.PYTHON_NEW_DATA_FILE_UPDATE,
     index,
     filename,
   };
 }
 
-export function deleteNewDataFile(index) {
+export function deletePythonNewDataFile(index) {
   return {
     type: actionTypes.PYTHON_NEW_DATA_FILE_DELETE,
     index,
   };
 }
 
-export function deleteExistingDataFile(filename, toDelete) {
+export function deletePythonExistingDataFile(filename, toDelete) {
   return {
     type: actionTypes.PYTHON_EXISTING_DATA_FILE_DELETE,
     filename,
     toDelete,
   };
 }
+
+export function updateCppCodeBlock(field, newValue) {
```

Comment:
Also i think that it is unlikely that i will need to change the editor actions to suit a specific language, and that the package itself should be designed to fit these same set of actions (since they are quite generic), instead of the other way around.